framework/web/UrlRule.php

```diff
@@ -216,6 +221,145 @@ public function init()
     }
 
     /**
+     * Initializes normalizer configuration
+     * @param UrlManager $manager the URL manager
+     * @throws \Exception if config is not understood
+     */
+    protected function initNormalizer($manager) {
+        $normalize = $this->normalize;
+
+        $configNoAction = [
+            // whether to collapse multiple slashes to single
+            'collapse-slashes' => false,
+            // remove trailing slash if it exists
+            'remove-trailing-slash' => false,
+            // add trailing slash if there is no one
+            'add-trailing-slash' => false,
```

Comment:
This two settings can be changed to one: `'trailing-slash' => true|false|null`.
* `true` - add trailing slash
* `false` - remove trailing slash
* `null` - do nothing